validator/evaluation/scoring.py

```diff
@@ -274,39 +274,73 @@ def _create_failed_miner_result(hotkey: str) -> MinerResults:
+    Docker evaluations with an exception will return the Exception for the repo."""
     evaluation_params = {
         "file_format": FileFormat.JSON,
         "original_model": task.model_id,
-        "model": submission_repo,
+        "models": submission_repos,
         "dataset_type": dataset_type,
         "gpu_ids": gpu_ids,
     }
 
     assert task.synthetic_data is not None, "Synthetic data shouldn't be none"
     assert task.test_data is not None, "Test data shouldn't be none"
+
     logger.info("Starting synth evaluation")
     synthetic_data_filepath = await download_s3_file(task.synthetic_data)
-    synth_eval_result = await run_evaluation_docker(dataset=synthetic_data_filepath, **evaluation_params)
+    synth_eval_results = await run_evaluation_docker(dataset=synthetic_data_filepath, **evaluation_params)
 
-    if not synth_eval_result.is_finetune:
-        return (
-            EvaluationResult(is_finetune=False, eval_loss=0.0, perplexity=0.0),
-            EvaluationResult(is_finetune=False, eval_loss=0.0, perplexity=0.0),
+    results: dict[str, tuple[EvaluationResult, EvaluationResult] | Exception] = {}
+
+    # First get synthetic results and identify finetuned models
+    finetuned_repos = []
+    for repo in submission_repos:
+        if isinstance(synth_eval_results.get(repo), Exception):
+            results[repo] = synth_eval_results[repo]
+            continue
+
+        synth_result = synth_eval_results[repo]
+        if not synth_result.is_finetune:
+            results[repo] = (
+                EvaluationResult(is_finetune=False, eval_loss=0.0, perplexity=0.0),
+                EvaluationResult(is_finetune=False, eval_loss=0.0, perplexity=0.0)
+            )
+        else:
+            finetuned_repos.append(repo)
+
+    # Only evaluate test data for finetuned models that succeeded synthetic evaluation
+    if finetuned_repos:
+        test_data_filepath = await download_s3_file(task.test_data)
+        test_eval_results = await run_evaluation_docker(
+            dataset=test_data_filepath,
+            models=finetuned_repos,
+            **{k: v for k, v in evaluation_params.items() if k != 'models'}
         )
 
-    test_data_filepath = await download_s3_file(task.test_data)
-    test_eval_result = await run_evaluation_docker(dataset=test_data_filepath, **evaluation_params)
+        # Update results with test evaluation
+        for repo in finetuned_repos:
+            if isinstance(test_eval_results.get(repo), Exception):
+                results[repo] = test_eval_results[repo]
+            else:
+                results[repo] = (synth_eval_results[repo], test_eval_results[repo])
+
+    # Add exceptions for any repos not yet processed
+    for repo in submission_repos:
+        if repo not in results:
+            results[repo] = Exception("Evaluation failed to complete")
 
-    return synth_eval_result, test_eval_result
+    return results
 
 
 async def _clear_up_s3(file_paths: list[str]) -> None:
```

Comment:
Shouldn't be in this file porfa

Reply:
I'm not sure which function you're referring to

Reply:
_clear_up_s3